src/seedu/addressbook/data/person/Address.java

```diff
@@ -35,24 +32,4 @@ public static boolean isValidAddress(String test) {
         return test.matches(ADDRESS_VALIDATION_REGEX);
     }
```

Comment:
The constructor and validation method are shared by all the classes, so better to abstract them as well.